playbooks/roles/start/tasks/wait_for_zowe.yml

```diff
@@ -66,25 +70,31 @@
   until: _result.status == 200
   retries: "{{ wait_for_zowe_service_retries | int }}"
   delay: "{{ wait_for_zowe_service_delay | int }}"
+  # if zowe_zos_host and zowe_external_domain_name are not same, zlux could be
+  # started off z/OS and then we couldn't test right now
+  when: zowe_zos_host == zowe_external_domain_name and "jobs-api" in zowe_launch_components
   delegate_to: localhost
   # hide log to avoid exposing zowe_test_user and zowe_test_password
   no_log: True
 
 - name: Wait for APIML port {{ zowe_apiml_gateway_port }} to be available
   uri:
-    url: "https://{{ zowe_external_domain_name }}:{{ zowe_apiml_gateway_port }}"
+    url: "https://{{ zowe_zos_host }}:{{ zowe_apiml_gateway_port }}"
     follow_redirects: none
     method: GET
     validate_certs: false
   register: _result
   until: _result.status == 200
   retries: "{{ wait_for_zowe_service_retries | int }}"
   delay: "{{ wait_for_zowe_service_delay | int }}"
+  # if zowe_zos_host and zowe_external_domain_name are not same, zlux could be
+  # started off z/OS and then we couldn't test right now
+  when: zowe_zos_host == zowe_external_domain_name and "api-mediation" in zowe_launch_components
```

Comment:
Please change to `when: zowe_zos_host == zowe_external_domain_name and ("gateway" in zowe_launch_components or zowe_launch_components == "")`